vivarium/simulator/states.py

```diff
@@ -0,0 +1,316 @@
+    exists = jnp.concatenate((existing_agents, existing_objects), dtype=int)
+
+    # TODO: Why is momentum set to none ?
+    return NVEState(
+        position=RigidBody(center=positions, orientation=orientations),
+        momentum=None,
+        force=RigidBody(center=jnp.zeros((n_entities, 2)), orientation=jnp.zeros(n_entities)),
+        mass=RigidBody(center=jnp.full((n_entities, 1), mass_center), orientation=jnp.full((n_entities), mass_orientation)),
+        entity_type=entity_types,
+        entity_idx = jnp.array(list(range(n_agents)) + list(range(n_objects))),
+        diameter=jnp.full((n_entities), diameter),
+        friction=jnp.full((n_entities), friction),
+        exists=exists
+        )
+
+
+# Could implement it as a static or class method
+def init_agent_state(
+        simulator_state: SimulatorState,
+        behavior: int = 1,
+        wheel_diameter: float = 2.,
+        speed_mul: float = 1.,
+        theta_mul: float = 1.,
+        prox_dist_max: float = 40.,
+        prox_cos_min: float = 0.,
+        color: str = "blue"
+        ) -> AgentState:
+    """
+    Initialize agent state with given parameters
+    """
+    n_agents = simulator_state.n_agents[0]
+    # TODO : Allow to define custom list of behaviors, wheel_diameters ... (in fact for all parameters)
+    # if the shape if just 1 value, assign it to all agents
+    # else, ensure you are given a list of arguments of size n_agents and transform it into a jax array
+
+    return AgentState(
+        nve_idx=jnp.arange(n_agents, dtype=int),
+        prox=jnp.zeros((n_agents, 2)),
+        motor=jnp.zeros((n_agents, 2)),
+        behavior=jnp.full((n_agents), behavior),
+        wheel_diameter=jnp.full((n_agents), wheel_diameter),
+        speed_mul=jnp.full((n_agents), speed_mul),
+        theta_mul=jnp.full((n_agents), theta_mul),
+        proxs_dist_max=jnp.full((n_agents), prox_dist_max),
+        proxs_cos_min=jnp.full((n_agents), prox_cos_min),
+        color=jnp.tile(_string_to_rgb(color), (n_agents, 1))
+    )
+
+
+def init_object_state(
+        simulator_state: SimulatorState,
+        color: str = "red"
+        ) -> ObjectState:
+    """
+    Initialize object state with given parameters
+    """
+    n_agents, n_objects = simulator_state.n_agents[0], simulator_state.n_objects[0]
+    start_idx, stop_idx = n_agents, n_agents + n_objects
+    objects_nve_idx = jnp.arange(start_idx, stop_idx, dtype=int)
+    return  ObjectState(
+        nve_idx=objects_nve_idx,
+        color=jnp.tile(_string_to_rgb(color), (n_objects, 1))
+    )
+
+
+def init_state(
+        simulator_state: SimulatorState,
+        agents_state: AgentState,
+        objects_state: ObjectState,
+        nve_state: NVEState
+        ) -> State:
+  
+    return State(
+        simulator_state=simulator_state,
+        agent_state=agents_state,
+        object_state=objects_state,
+        nve_state=nve_state
+    )
+    
```

Comment:
A general comment to discuss:

If I understand well, with this PR a user will be able to define a custom state more easily from a run script, which is nice indeed. But if the objective is to simplify the life of the use, I am wondering if we shouldn't abstract the distinction between the NVEState and the Agent/ObjectState. How about to just having a function where a user can just enter what he wants (e.g. how many agents/objects, which positions etc ...) and then the function will deal with how to construct the State, hiding this complexity to the user.